server/models/Payment.js

```diff
@@ -0,0 +1,43 @@
+const mongoose = require('mongoose');
+const Sitter = required('./Sitter.js');
+
+const Payment = new mongoose({
+  userId: {
+    type: mongoose.Schema.Types.ObjectId,
+    required: true,
+    ref: 'User',
+  },
+  sitterId: {
+    type: mongoose.Schema.Types.ObjectId,
+    required: true,
+    ref: 'Sitter',
+  },
+  rate: {
+    type: Number,
+    required: true,
+    set: getRate,
+  },
+  hoursOfService: {
+    type: Number,
+    required: true,
+  },
+  totalPayment: {
+    type: Number,
+    required: true,
+    set: payTotalOf,
+  },
+  customerId: {
+    type: String,
+    required: true,
+  },
+});
+
+const getRate = async (sitterId) => {
+  const sitter = await Sitter.findById(sitterId);
+  return sitter.rate;
+};
+
+const payTotalOf = (rate, hoursOfService) => {
+  const processingFee = 5;
+  retrun(rate * hoursOfService + processingFee);
+};
```

Comment:
Let's look at getters in Mongoose for this kind of calculation.

Reply:
fixed that in an old branch its good in the main branch
will address it here (i did all those fixes